apps/dashboard/src/components/workflow-editor/steps/delay/delay-configure.tsx

```diff
@@ -0,0 +1,136 @@
+import { Separator } from '@/components/primitives/separator';
+import { SidebarContent } from '@/components/side-navigation/sidebar';
+import { CommonFields } from '@/components/workflow-editor/steps/common-fields';
+import { useBlocker, useParams } from 'react-router-dom';
+import { flattenIssues } from '@/components/workflow-editor/step-utils';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useStepEditorContext } from '@/components/workflow-editor/steps/hooks';
+import { buildDefaultValues, buildDynamicZodSchema } from '@/utils/schema';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { Form } from '@/components/primitives/form/form';
+import { useWorkflowEditorContext } from '@/components/workflow-editor/hooks';
+import debounce from 'lodash.debounce';
+import { z } from 'zod';
+import { TimeUnitEnum } from '@novu/shared';
+import { useUpdateWorkflow } from '@/hooks/use-update-workflow';
+import { showToast } from '@/components/primitives/sonner-helpers';
+import { ToastIcon } from '@/components/primitives/sonner';
+import { UnsavedChangesAlertDialog } from '@/components/unsaved-changes-alert-dialog';
+import merge from 'lodash.merge';
+import { DelayAmount } from '@/components/workflow-editor/steps/delay/delay-amount';
+
+const TOAST_CONFIG = {
+  position: 'bottom-left' as const,
+  classNames: { toast: 'ml-10 mb-4' },
+};
+
+const delayControlsSchema = z
+  .object({
+    type: z.enum(['regular']).default('regular'),
+    amount: z.number(),
+    unit: z.nativeEnum(TimeUnitEnum),
+  })
+  .strict();
+
+export const DelayConfigure = () => {
+  const { stepSlug = '' } = useParams<{ workflowSlug: string; stepSlug: string }>();
+  const { step, refetch } = useStepEditorContext();
```

Comment:
we have 2 types of step :

```jsx
1. const { step } = useStep()
2. const { step } = useStepEditorContext() 
```
The 1st one refetches when workflow is updated (desired I would say), the 2nd one doesn't until you call `refetch()`.  Could we have this behaviour here as well or its intended?